### DISCUSSION TITLE
Again on Technics USB Audio device in Technics SU-G700M2

### DISCUSSION THREAD

**Discussion by Argo16x** [rern/rAudio#974] [Bug Reports]
As stated in the inherent feature request, rAudio works with Technics SU-G700M2 USB input digital connection (except for .dsf files) but ALSA aplay fails setting the card parameters at first try (at startup or anytime you play a file with a different sample rate) while works flawlessly at 2nd try. So everytime I have to push play, wait for the button becoming pause and again play, and push it again. This for each change of sample rate in the playlist file. 
I'm almost total ignorant on Linux/MPD/ALSA but I'like to know, if possible, how to implement, as a workaround,  a way that player calls ALSA (or MPD) twice instead of one time for each play function.

Thank you

**Comment by Argo16x** [comment]
@rern ?

**Reply by rern** [reply]
It seems like a driver issue.
I think a HAT for S/PDIF optical/coaxial output would be better than a workaround.

**Reply by Argo16x** [reply]
@rern 

Yes it probably is and I opened a bug with the kernel maintainers.
I though about using an Hifiberry Digi Pro or an Allo DigiOne hat but I'd lose some of the high sample capabilities of the USB interface. I agree that any sound difference would not be perceptible but I would have to convert all those files.... If possible, and reasonable could you suggest me what to modify and how to have actually the player call mpd  twice for each play command?

Thank you.
